framework/Optimizers/SPSA.py

```diff
@@ -151,8 +172,12 @@ def localGenerateInput(self,model,oldInput):
                 if var not in self.gradDict['pertPoints'][traj][ind].keys():
                   p1 = np.asarray([varK[var]+ck*delta[varID]*1.0]).reshape((1,))
                   p2 = np.asarray([varK[var]-ck*delta[varID]*1.0]).reshape((1,))
+                  #sanity check: p1 != p2
```

Comment:
It's kind of secretly preventing another bug too, but I'm still working on that in the extended branch...